TELCODOCS-2197: Add support for user defined tolerations to modules (#89568)
openshift/openshift-docs: modules/kmm-using-tolerations-for-kernel-module-scheduling.adoc

// Module included in the following assemblies:
//
// * hardware_enablement/kmm-kernel-module-management.adoc

:_mod-docs-content-type: CONCEPT
[id="kmm-using-tolerations-for-kernel-module-scheduling_{context}"]
= Using tolerations for kernel module scheduling

> There are circumstances where you will need to evacuate workloads on a node before upgrading a kernel module, which you can do through taints. 

Author: Made this change.

> A taint allows a node to refuse a pod to be scheduled unless that pod has a matching toleration.

Reviewer: ```suggestion
You can use a taint to schedule only pods that contain a matching toleration on the node.
```

 
Avoid using "allow" for inanimate objects: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#a

> Avoid stating that inanimate objects grant abilities to people, as in “the product allows you to…”. Whenever possible, use a direct, user-focused alternative such as “you can use the product to…” or “with this product, users can…”.

Author: Made this change.

> However, users also need to set tolerations to allow KMM pods that run housekeeping operations, such as kernel module upgrades. The tolerations must match the taint that gets added to the nodes.

Reviewer: ```suggestion
However, you also need to set tolerations to allow Kernel Module Management (KMM) pods that run housekeeping operations, such as kernel module upgrades. The tolerations must match the taint that is added to the nodes.
```

KMM is being used for the first time here in the module.

> You can create user-defined tolerations to kernel modules to schedule selected KMM pods on a cordoned node. For example, during a device driver upgrade you can specify that a node is to be cordoned while, at the same time, you can run schedulable housekeeping pods that carry out the driver upgrades. 

Reviewer: I am thinking of the following rewrites:

- 1. "...during a device driver upgrade you can cordon a node, at the same time, you can run housekeeping pods to upgrade a device driver."
- 2. "...during a device driver upgrade you cordon a node, at the same time, you can run housekeeping pods that perform the driver upgrades"

Author: Made this change (ii.)

// Module included in the following assemblies:
//
// * hardware_enablement/kmm-kernel-module-management.adoc

:_mod-docs-content-type: CONCEPT
[id="kmm-using-tolerations-for-kernel-module-scheduling_{context}"]
= Using tolerations for kernel module scheduling

There are circumstances where you will need to evacuate workloads on a node before upgrading a kernel module, which you can do through taints. 
A taint allows a node to refuse a pod to be scheduled unless that pod has a matching toleration.

However, users also need to set tolerations to allow KMM pods that run housekeeping operations, such as kernel module upgrades. The tolerations must match the taint that gets added to the nodes.

You can create user-defined tolerations to kernel modules to schedule selected KMM pods on a cordoned node. For example, during a device driver upgrade you can specify that a node is to be cordoned while, at the same time, you can run schedulable housekeeping pods that carry out the driver upgrades. 

> ModuleSpec is used to carry the tolerations to the KMM pods that are used during pod creation.

Reviewer 2: ```suggestion
`ModuleSpec` field is used to carry the tolerations to the KMM pods that are used during pod creation.
```

Author: Made changes.

> 

Reviewer 2: extra lines here

Author: Deleted lines.
